pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -219,16 +219,7 @@ self: super: {
   # Needs older version of QuickCheck.
   these_0_7_6 = doJailbreak super.these_0_7_6;
 
-  # dontCheck: Can be removed once https://github.com/haskell-nix/hnix/commit/471712f is in (5.2 probably)
-  #   This is due to GenList having been removed from generic-random in 1.2.0.0
-  # doJailbreak: Can be removed once https://github.com/haskell-nix/hnix/pull/329 is in (5.2 probably)
-  #   This is due to hnix currently having an upper bound of <0.5 on deriving-compat, works just fine with our current version 0.5.1 though
-  # Does not support recent versions of "these".
-  # https://github.com/haskell-nix/hnix/issues/514
-  hnix =
-    generateOptparseApplicativeCompletion "hnix" (
-      dontCheck (doJailbreak (super.hnix.override { these = self.these_0_7_6; }))
-    );
+  hnix = dontCheck (super.hnix);
```

Comment:
Could you do two things here:

- Remove `hnix` from the broken packages section in the `pkgs/development/haskell-modules/configuration-hackage2nix.yaml`, since it is building now.
- Add a comment here why `dontCheck` is needed (if it makes sense).

Reply:
Yep, `hnix` can not pass its testing suite.

And btw, maybe would add that  `gen*Optparse*Completion` plainly does not work - it falls the build.

I thought I've seen it is broken, but on yesterday's search, I did not find that, thanks for pointing it out. I am still learning infrastructure.

Reply:
What was added:
  * `hnix` doc why `dontCheck`.
  * `hnix` doc why no `gen*Optparse*Completion`
  * rm `these ==0.7.6` from `configuration-hackage2nix.yaml` and overrides for it in `configuration-common.nix `, since this package version was used only by old `hnix` version

Reply:
> And btw, maybe would add that gen*Optparse*Completion plainly does not work - it falls the build.

I tried 

```diff
diff --git a/pkgs/development/haskell-modules/configuration-common.nix b/pkgs/development/haskell-modules/configuration-common.nix
index 8409ff2778e..72f6d951ada 100644
--- a/pkgs/development/haskell-modules/configuration-common.nix
+++ b/pkgs/development/haskell-modules/configuration-common.nix
@@ -219,7 +219,7 @@ self: super: {
   # 2020-06-05: HACK: `generateOptparseApplicativeCompletion` does not work,
   # build fails - option disabled
   # 2020-06-05: HACK: does not passes own build suite - `dontCheck`
-  hnix = dontCheck (super.hnix);
+  hnix = generateOptparseApplicativeCompletion "hnix" (dontCheck (super.hnix));
 
   # Fails for non-obvious reasons while attempting to use doctest.
   search = dontCheck super.search;
```

and it appears to build.

What sort of error did you get when you tried this?

Reply:
There was an error during the generation of the completion.

Maybe it is because I started with this package first, so before its clean-up, and as so package used for example `these` old vertion.

Reply:
And indeed, the current upstream code fails to generate complettions:

```
Test suite logged to: dist/test/hnix-0.8.0-hnix-tests.log
1 of 1 test suites (1 of 1 test cases) passed.
haddockPhase
installing
Installing library in /nix/store/i5rk2nl0yaa2w4rbzqxmlv53jrambzyg-hnix-0.8.0/lib/ghc-8.8.2/x86_64-linux-ghc-8.8.2/hnix-0.8.0-Je1zkJQRPx69EBuUzc9dwC
/nix/store/ba8fhbmvfyh439606lni91y420yflmlc-stdenv-linux/setup: line 85: /nix/store/i5rk2nl0yaa2w4rbzqxmlv53jrambzyg-hnix-0.8.0/bin/hnix-0.8.0: No such file or directory
builder for '/nix/store/34kid81jvakazykkfsv7p8b2qpgx0b9i-hnix-0.8.0.drv' failed with exit code 127
error: build of '/nix/store/34kid81jvakazykkfsv7p8b2qpgx0b9i-hnix-0.8.0.drv' failed
```

So I would recommend for now leave it as is, until it would be figured-out.

Reply:
Oh, I figured it out. `developPackage.name` is not a `package.name`, it adds a version to it.

Reply:
Enabled completions.

Reply:
I would be glad when completions are tested in the upstream CI.